worker/jobTypes/githubJob.js

```diff
@@ -132,23 +132,51 @@ class GitHubJobClass {
 
     try {
       const exec = workerUtils.getExecPromise();
-      
-      const pullRepoCommands = [
-        `cd repos/${this.getRepoDirName()}`,
-      ];
+      const pullRepoCommands = [`cd repos/${this.getRepoDirName()}`];
 
       // if commit hash is provided, use that
       if (currentJob.payload.newHead) {
-        pullRepoCommands.push(...[
-          `git checkout ${currentJob.payload.newHead}`,
-          `git checkout -b ${currentJob.payload.branchName}`,
-          `git pull origin ${currentJob.payload.branchName}`,
-        ]);
-      } else {
-        pullRepoCommands.push(...[
+        const commitCheckCommands = [
+          `cd repos/${this.getRepoDirName()}`,
+          `git fetch`,
           `git checkout ${currentJob.payload.branchName}`,
-          `git pull origin ${currentJob.payload.branchName}`,
-        ]);
+          `git branch ${currentJob.payload.branchName} --contains ${currentJob.payload.newHead}`
+        ];
+
+        const result = await exec(commitCheckCommands.join('&&'));
+        const returnObject = {};
+        
+        if (!result.stdout.includes(`* ${currentJob.payload.branchName}`)) {
+          const err = new Error(
+            `Specified commit does not exist on ${currentJob.payload.branchName} branch`
+          );
+          logger.save(
+            `${'(BUILD)'.padEnd(
+              15
+            )} failed. The specified commit does not exist on ${
+              currentJob.payload.branchName
+            } branch.`
+          );
+          return new Promise(function(resolve, reject) {
+            reject(err);
+          });
+        }
+
+        pullRepoCommands.push(
+          ...[
+            `git checkout ${currentJob.payload.branchName}`,
+            `git pull origin ${currentJob.payload.branchName}`,
+            `git checkout ${currentJob.payload.newHead}`
+          ]
+        );
+
+      } else {
+        pullRepoCommands.push(
+          ...[
+            `git checkout ${currentJob.payload.branchName}`,
+            `git pull origin ${currentJob.payload.branchName}`
+          ]
+        );
       }
 
       await exec(pullRepoCommands.join(' && '));
```

Comment:
let's check for standard error here to. In general, if we are running exec we need to check output for SE and report back/fail properly.